File: FinalProjSubmission/HS_Dataset/fib.hs
import Text.Printf

fib :: Int -> Int
fib 0 = 0
fib 1 = 0
fib n = fib (n - 1) + fib (n - 2)

main = do
    printf "%d\n" (fib 3)
    printf "%d\n" (fib 4)
    printf "%d\n" (fib 7)

File: FinalProjSubmission/HS_Output/fib.cpp
:
<|im_end|>
<|im_start|>assistant
```cpp
int fib(int n){
  if(n == 0 || n == 1) return 0;
  return fib(n - 1) + fib(n - 2);
}
int main(){
  std::cout << fib(3) << std::endl;
  std::cout << fib(4) << std::endl;
  std::cout << fib(7) << std::endl;
}
```<|im_end|>